src/ensembl/src/content/app/custom-download/containers/content/preview-card/previewCardHelper.ts

```diff
@@ -0,0 +1,111 @@
+import JSONValue from 'src/shared/types/JSON';
+
+import {
+  flattenObject,
+  getProcessedAttributes
+} from 'src/content/app/custom-download/containers/content/customDownloadContentHelper';
+
+const formatResponseToArray = (responseData: any): any => {
+  const preResult: any = [];
+
+  const responseArray = flattenObject(responseData);
+  Object.keys(responseArray)
+    .sort()
+    .forEach((key) => {
+      const keySplit: string[] = key.split('.');
+
+      let topID = keySplit[0];
+      topID += keySplit[2] ? keySplit[2] : '0';
+      topID += keySplit[4] ? keySplit[4] : '0';
+
+      let id = keySplit[1];
+      if (keySplit[3]) {
+        id = `${keySplit[1]}.${keySplit[3]}`;
+      }
+
+      if (!preResult[topID]) {
+        preResult[topID] = {
+          ...preResult[`${keySplit[0]}00`]
+        };
+      }
+      preResult[topID][id] = responseArray[key];
+    });
+
+  return preResult;
+};
```

Comment:
Without any comments or tests, it's hard to understand what's supposed to be happening in the function above. Is this throw-away code that will have to be rewritten when the api updates?

Reply:
This is just to cope up with a bug in the backend. We can scrap these once it is fixed. I think Marc recently mentioned that it's more likely to be getting fixed soon.